### ISSUE TITLE
Inconsistency with criterion.sizeAverage semantics

### ISSUE THREAD

**Issue by davidBelanger** [torch/nn#230]
In ClassNLLCriterion, if self.sizeAverage is true, then the loss and gradient get divided by the number of examples in the minibatch. 

However, MultiMarginCriterion (both in nn and cunn), if self.sizeAverage is true, the  loss and gradient get divided by the number of target labels. 

I'm wondering what the convention is supposed to be. Dividing by the size of the minibatch makes a lot more sense to me. Otherwise the learning rate will interact with the minibatch size in weird ways. Also, in MultiMarginCriterion.cu, the nframe variable is declared, but never used, so this makes me think that we should have been dividing by nframe, rather than dim. 

This sounds like a bug with MultiMarginCriterion. I can make a PR to fix it, but I want to make sure I'm interpreting things correctly first.

**Comment by nicholas-leonard** [comment]
Yes I agree that a PR should be made.

**Comment by davidBelanger** [comment]
Ok. Just to clarify, this would require changing the definition of the loss in https://github.com/torch/nn/blob/master/doc/criterion.md#nn.MultiMarginCriterion
Should I worry about suddenly changing its definition on people?

**Comment by nicholas-leonard** [comment]
Not many people use this Criterion AFAIK. If you are motivated, you could make it backwards compatible with previously serialized MultiMarginCriterions. You can look at nn.Dropout for an example of how to do this.
